src/utility/random.cpp

```diff
@@ -62,30 +77,39 @@ static std::mt19937& get_twister()
     return *twister;
 }
 
-uint64_t pseudo_random()
+uint64_t pseudo_random::next()
 {
-    return pseudo_random(0, max_uint64);
+    return next(0, max_uint64);
 }
 
-uint64_t pseudo_random(uint64_t begin, uint64_t end)
+uint64_t pseudo_random::next(uint64_t begin, uint64_t end)
 {
     std::uniform_int_distribution<uint64_t> distribution(begin, end);
     return distribution(get_twister());
 }
 
-void pseudo_random_fill(data_chunk& chunk)
+template<class Container>
+void pseudo_random::fill(Container& out)
 {
     // uniform_int_distribution is undefined for sizes < 16 bits.
     std::uniform_int_distribution<uint16_t> distribution(0, max_uint8);
+    auto& twister = pseudo_random::get_twister();
+
+    std::transform(out.begin(), out.end(), out.begin(),
+        [&distribution, &twister](uint8_t byte) {
+            return distribution(twister); });
+}
 
-    for (auto& byte: chunk)
-        byte = static_cast<uint8_t>(distribution(get_twister()));
+asio::duration pseudo_randomize(const asio::duration& expiration,
+    uint8_t ratio)
+{
+    return pseudo_random::duration(expiration, ratio);
 }
 
 // Randomly select a time duration in the range:
 // [(expiration - expiration / ratio) .. expiration]
 // Not fully testable due to lack of random engine injection.
```

Comment:
Note that by making the engine a member it can be overridden for test, making these methods now testable.